src/shims/posix/fs.rs

```diff
@@ -28,7 +28,11 @@ trait FileDescriptor : std::fmt::Debug {
     fn read<'tcx>(&mut self, communicate_allowed: bool, bytes: &mut [u8]) -> InterpResult<'tcx, io::Result<usize>>;
     fn write<'tcx>(&mut self, communicate_allowed: bool, bytes: &[u8]) -> InterpResult<'tcx, io::Result<usize>>;
     fn seek<'tcx>(&mut self, communicate_allowed: bool, offset: SeekFrom) -> InterpResult<'tcx, io::Result<u64>>;
-    fn close<'tcx>(&mut self, communicate_allowed: bool) -> InterpResult<'tcx, io::Result<i32>>;
+    fn close<'tcx>(self: Box<Self>, _communicate_allowed: bool) -> InterpResult<'tcx, io::Result<i32>> {
+        throw_unsup_format!("stdin/stdout/stderr cannot be closed(not supported by Rust)");
```

Comment:
```suggestion
        throw_unsup_format!("stdin/stdout/stderr cannot be closed");
```
But also, I think I'd prefer this to be in the stdin/stdout/stderr impls; having it in the default impl seems a bit odd. That also will allow the message to be a bit more specific.

Reply:
Moved these to stdin/out/err impls